render: use elk edge label positioning

gavin-ts opened the pull request:
## Summary

Elk layout positions edge labels on the side, but we position labels in the middle and this could lead to unexpected container padding. We now configure elk to position edge labels the same way so that it doesn't add this extra padding.

## Details
- fixes #482
- configure elk to place edge labels on the edge
- adds e2e test


### e2e report changes with new test

![_Users_gavinnishizawa_github_repos_d2_out_e2e_report html (4)](https://user-images.githubusercontent.com/85081687/208786343-a15a7ddc-9772-444b-84be-bea6ed9aca63.png)

alixander commented:
oh interesting, ELK naturally just placed the labels to the left?

alixander commented:
can we tune it to... not always be left? https://www.eclipse.org/elk/reference.html

gavin-ts commented:
> oh interesting, ELK naturally just placed the labels to the left?

default is their `SMART_DOWN` option, which usually places it on the left side. [ref](https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-edgeLabels-sideSelection.html)

> can we tune it to... not always be left? https://www.eclipse.org/elk/reference.html

it doesn't look like we can put it on the edge itself, but I added some padding and switched it to the `SMART_UP` option which usually places labels on the right instead.

alixander commented:
they do allow it: https://www.eclipse.org/elk/reference/options/org-eclipse-elk-edgeLabels-inline.html

gavin-ts commented:
> they do allow it: https://www.eclipse.org/elk/reference/options/org-eclipse-elk-edgeLabels-inline.html

nice it looks more consistent with the other layouts in the center position

<img width="1262" alt="Screen Shot 2022-12-20 at 3 36 13 PM" src="https://user-images.githubusercontent.com/85081687/208786648-c2a131a3-112d-4bf8-858b-3ef1f9d7160e.png">
